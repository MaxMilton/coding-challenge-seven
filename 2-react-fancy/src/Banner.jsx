import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ResponsiveImg from './ResponsiveImg';

export const Wrapper = styled.div`
  position: relative;
  display: inline-block;
`;

export const TileImage = styled(ResponsiveImg)`
  display: block;
`;

export const Overlay = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  width: 100%;
  padding: 1rem;
  box-sizing: border-box;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
`;

export const NetworkLogo = styled(ResponsiveImg)`
  width: auto;
  height: 4.6rem;
`;

export const Title = styled.h3`
  margin: 0 0 0 1rem;
  color: rgb(255, 255, 255);
  font-family: sans-serif;
  font-size: 2.2rem;
`;

const Banner = ({
  className,
  tileX1,
  tileX2,
  title,
  logoX1,
  logoX2,
  logoAlt,
}) => (
  <Wrapper className={className}>
    <TileImage imgX1={tileX1} imgX2={tileX2} altText={title} />

    <Overlay>
      <NetworkLogo imgX1={logoX1} imgX2={logoX2} altText={logoAlt} />

      <Title>
        {title}
      </Title>
    </Overlay>
  </Wrapper>
);

Banner.propTypes = {
  tileX1: PropTypes.string.isRequired,
  tileX2: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  logoX1: PropTypes.string,
  logoX2: PropTypes.string,
  logoAlt: PropTypes.string,
};

Banner.defaultProps = {
  className: '',
  logoX1: '../logo.png',
  logoX2: '../logo@x2.png',
  logoAlt: '7plus',
};

export default Banner;
