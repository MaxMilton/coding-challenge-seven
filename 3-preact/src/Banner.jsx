import { h } from 'preact';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  position: relative;
  display: inline-block;
`;

const TileImage = styled.img`
  display: block;
`;

const Overlay = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  width: 100%;
  padding: 1rem;
  box-sizing: border-box;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
`;

const NetworkLogo = styled.img`
  width: auto;
  height: 4.6rem;
`;

const Title = styled.h3`
  margin: 0 0 0 1rem;
  color: rgb(255, 255, 255);
  font-family: sans-serif;
  font-size: 2.2rem;
`;

const Banner = ({
  className,
  tile,
  title,
  logo,
  logoAlt,
}) => (
  <Wrapper className={className}>
    <TileImage src={tile} alt={title} />

    <Overlay>
      <NetworkLogo src={logo} alt={logoAlt} />

      <Title>
        {title}
      </Title>
    </Overlay>
  </Wrapper>
);

Banner.propTypes = {
  tile: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  logo: PropTypes.string,
  logoAlt: PropTypes.string,
};

Banner.defaultProps = {
  className: '',
  logo: '../logo.png',
  logoAlt: '7plus',
};

export default Banner;
