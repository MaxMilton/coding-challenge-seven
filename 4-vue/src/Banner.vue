<template>
  <div class="banner">
    <img :src="tile" :alt="title" class="tile" />

    <div class="overlay">
      <img :src="logo" :alt="logoAlt" class="network" />

      <h3 class="title">
        {{ title }}
      </h3>
    </div>
  </div>
</template>

<script>
  export default {
    props: {
      tile: String,
      title: String,
      logo: {
        type: String,
        default: '../logo.png',
      },
      logoAlt: {
        type: String,
        default: '7plus',
      },
    },
  };
</script>

<style scoped>
  .banner {
    position: relative;
    display: inline-block;
  }

  .tile {
    display: block;
  }

  .overlay {
    position: absolute;
    bottom: 0;
    box-sizing: border-box;
    display: flex;
    align-items: center;
    width: 100%;
    padding: 1rem;
    background-color: rgba(0, 0, 0, 0.5);
  }

  .network {
    width: auto;
    height: 4.6rem;
  }

  .title {
    margin: 0 0 0 1rem;
    font-family: sans-serif;
    font-size: 2.2rem;
    color: rgb(255, 255, 255);
  }
</style>
