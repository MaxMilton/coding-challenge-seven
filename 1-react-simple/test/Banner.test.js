import React from 'react';
import renderer from 'react-test-renderer';
import Banner from '../src/Banner';

describe('Banner component', () => {
  it('prints errors with no props', () => {
    expect.assertions(1);
    const spy = jest.spyOn(console, 'error').mockImplementation(() => {});
    renderer.create(
      <Banner />,
    );
    expect(spy).toHaveBeenCalledTimes(2);
    spy.mockReset();
    spy.mockRestore();
  });

  it('renders correctly', () => {
    expect.assertions(2);
    const spy = jest.spyOn(console, 'error');
    const component = renderer.create(
      <Banner tile="../tile.jpg" title="Home And Away" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(spy).not.toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });
});
