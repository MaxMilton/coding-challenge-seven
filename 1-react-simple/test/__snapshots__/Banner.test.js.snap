// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `
<div
  className="sc-bdVaJa jCeSos"
>
  <img
    alt="Home And Away"
    className="sc-bwzfXH gbieAY"
    src="../tile.jpg"
  />
  <div
    className="sc-htpNat kuzZIM"
  >
    <img
      alt="7plus"
      className="sc-bxivhb kufJPE"
      src="../logo.png"
    />
    <h3
      className="sc-ifAKCX eVJwop"
    >
      Home And Away
    </h3>
  </div>
</div>
`;
